src/app/CommandSender.h

```diff
@@ -53,96 +53,158 @@ namespace app {
 class CommandSender final : public Messaging::ExchangeDelegate
 {
 public:
-    // CommandSender::Callback::OnResponse is public SDK API, so we cannot break source
-    // compatibility for it. To allow for additional values to be added at a future time
-    // without constantly changing the function's declaration parameter list, we are
-    // defining the struct AdditionalResponseData and adding that to the parameter list
-    // to allow for future extendability.
-    struct AdditionalResponseData
-    {
-        Optional<uint16_t> mCommandRef;
-    };
-
+    /**
+     * @brief Legacy callbacks for CommandSender
```

Comment:
As a followup PR (to make the review simpler, though in practice most of this class is ending up looking like it was added in this PR anyway): please move this legacy class down below all the actual non-deprecated public APIs, so it's not the first thing you see when you look at CommandSender.

Reply:
Added todo comment and added to my issue tracker
https://github.com/project-chip/connectedhomeip/issues/30453